ambry-cloud/src/main/java/com/github/ambry/cloud/HelixVcrCluster.java

```diff
@@ -167,4 +171,22 @@ public void close() {
     manager.disconnect();
     helixAdmin.close();
   }
+
+  /**
+   * Get the name of the state model for the factory class.
+   * @param stateModelFactoryClass name of the {@link org.apache.helix.participant.statemachine.StateModelFactory} class.
+   * @return state model name corresponding to the factor class.
+   */
+  private String getStateModelNameFromFactory(String stateModelFactoryClass) {
+    if (stateModelFactoryClass.equals(OnlineOfflineHelixVcrStateModelFactory.class.getName())) {
+      return OnlineOfflineSMD.name;
+    }
+    if (stateModelFactoryClass.equals(LeaderStandbyHelixVcrStateModelFactory.class.getName())) {
+      return LeaderStandbySMD.name;
+    }
+    throw new IllegalArgumentException(String.format(
+        "Illegal vcr helix state model factory class: " + stateModelFactoryClass + ". Should be one of"
+            + OnlineOfflineHelixVcrStateModelFactory.class.getName() + ", "
+            + LeaderStandbyHelixVcrStateModelFactory.class.getName()));
```

Comment:
[An alternative but totally optional]
Since we decide to use two factories, we can make a abstract factory `VcrStateModelFactory` which extends `StateModelFactory` and make two state model factories extend it. Then this method can be replaced by adding state model name to each StateModelFactory. Thus, line149~151 can be changed to:
```
VcrStateModelFactory stateModelFactory = Utils.getObj(cloudConfig.vcrHelixStateModelFactoryClass, this);
manager.getStateMachineEngine()
        .registerStateModelFactory(stateModelFactory.modelName, stateModelFactory);
```
And this method can be removed.